packages/components/src/TemporaryAddressChange/Component.purs

```diff
@@ -29,14 +30,15 @@ import React.Basic.DOM.Events (preventDefault)
 import React.Basic.Events (handler, handler_)
 
 type State =
-  { startDate     :: Maybe DateTime
-  , minStartDate  :: Maybe DateTime
-  , endDate       :: Maybe DateTime
-  , minEndDate    :: Maybe DateTime
-  , streetAddress :: Maybe String
-  , zipCode       :: Maybe String
-  , countryCode   :: Maybe String
-  , temporaryName :: Maybe String
+  { startDate      :: Maybe DateTime
+  , minStartDate   :: Maybe DateTime
+  , endDate        :: Maybe DateTime
+  , minEndDate     :: Maybe DateTime
+  , streetAddress  :: Maybe String
+  , zipCode        :: Maybe String
+  , countryCode    :: Maybe String
+  , temporaryName  :: Maybe String
+  , perpetualCheck :: Maybe Boolean
```

Comment:
I think we can get rid of the `Maybe` here as well. 

As for the name of this property, I would change it to something like `isIndefinite`.

Reply:
I like the name but this needs to be a Maybe as that is what `React.Basic.DOM.Events.targetChecked` returns

Reply:
We can assume that `Nothing` means  `false` here (what we actually do in other parts of the component already).

Seems you are already looking into the return value of `targedChecked`:
```purescript
onChange: handler targetChecked \maybeNewVal -> do
                self.setState _ { checked = fromMaybe false maybeNewVal }
                props.onChange maybeNewVal
```
What's missing is to change `onChange` to take `Boolean` instead of `Maybe Boolean`.

Reply:
Or handle the `Maybe` in this part
```purescript
onChange: \checked -> self.setState _ { perpetualCheck = checked }
```

Reply:
Sense this makes. cc742f8